event-schema/event_schema_test.go

```diff
@@ -1,13 +1,202 @@
-package event_schema_test
+package event_schema
 
 import (
+	"database/sql"
+	"encoding/json"
+	"log"
+	"os"
 	"testing"
 
-	. "github.com/onsi/ginkgo/v2"
-	. "github.com/onsi/gomega"
+	"github.com/jeremywohl/flatten"
+	"github.com/ory/dockertest/v3"
+	"github.com/rudderlabs/rudder-server/admin"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	migrator "github.com/rudderlabs/rudder-server/services/sql-migrator"
+	"github.com/rudderlabs/rudder-server/services/stats"
+	"github.com/rudderlabs/rudder-server/testhelper"
+	"github.com/rudderlabs/rudder-server/testhelper/destination"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/stretchr/testify/require"
 )
 
-func TestAlert(t *testing.T) {
-	RegisterFailHandler(Fail)
-	RunSpecs(t, "Alert Suite")
+var pgResource *destination.PostgresResource
+
+func TestMain(m *testing.M) {
```

Comment:
Can you please avoid using `TestMain` and simply do one of:
1. nested test like we do in `docker_test.go`
2. use ginkgo with contexts

If you use `TestMain` then it will be impossible to run a unit test in the same package without executing this code that is completely unrelated and it is creating actual Docker containers (i.e. psql).

Reply:
Good point, let me change it up a bit.